week7/answers.sql

```diff
@@ -1,3 +1,287 @@
+);
+
+CREATE TABLE items (
+  item_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  name VARCHAR(30) NOT NULL,
+  armor INT NOT NULL,
+  damage INT NOT NULL
+);
+
+CREATE TABLE inventory (
+  inventory_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  character_id INT UNSIGNED NULL,
+  item_id INT UNSIGNED NULL,
+  FOREIGN KEY (character_id) REFERENCES characters(character_id)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE,
+  FOREIGN KEY (item_id) REFERENCES items(item_id)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+);
+
+CREATE TABLE equipped (
+  equipped_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  character_id INT UNSIGNED NOT NULL,
+  item_id INT UNSIGNED NOT NULL,
+  FOREIGN KEY (character_id) REFERENCES characters(character_id)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE,
+  FOREIGN KEY (item_id) REFERENCES items(item_id)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+);
+
+CREATE VIEW character_items AS
+SELECT DISTINCT
+    c.character_id,
+    c.name AS character_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM
+    characters c
+INNER JOIN (
+    SELECT character_id, item_id
+    FROM inventory
+    UNION ALL
+    SELECT character_id, item_id
+    FROM equipped
+) all_items ON c.character_id = all_items.character_id
+INNER JOIN items i ON all_items.item_id = i.item_id
+  ORDER BY c.character_id, i.name;
+
+
+CREATE VIEW team_items AS
+SELECT DISTINCT
+    t.team_id,
+    t.name AS team_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM
+    teams t
+INNER JOIN
+    team_members tm ON t.team_id = tm.team_id
+INNER JOIN
+    inventory inv ON tm.character_id = inv.character_id
+INNER JOIN
+    equipped eq ON tm.character_id = eq.character_id
+INNER JOIN
+    items i ON i.item_id = COALESCE(inv.item_id, eq.item_id);
+
+
+
+
+DELIMITER $$
+
+CREATE FUNCTION armor_total(char_id INT UNSIGNED) 
+RETURNS INT
+READS SQL DATA
+BEGIN
+    -- Variables to store intermediate results
+    DECLARE char_armor INT;
+    DECLARE equipped_armor INT;
+    DECLARE total_armor INT UNSIGNED;
+
+    -- Get the base armor from character stats (if available)
+    SELECT armor INTO char_armor
+    FROM character_stats
+    WHERE character_id = char_id;
+
+    -- Sum the armor values of equipped items (if any)
+    SELECT SUM(i.armor) INTO equipped_armor
+    FROM equipped e
+    INNER JOIN items i ON e.item_id = i.item_id
+    WHERE e.character_id = char_id;
+
+    -- Add base stats and equipped armor
+    SET total_armor = char_armor + equipped_armor;
+
+    RETURN total_armor;
+END $$
+
+CREATE PROCEDURE attack(
+    IN target_character_id INT,
+    IN equipped_item_id INT
+)
+BEGIN
+    DECLARE actual_item_id INT;
+    DECLARE equipped_item_damage INT;
+    DECLARE target_character_armor INT;
+    DECLARE target_character_health INT;
+
+    SELECT item_id INTO actual_item_id
+    FROM equipped
+    WHERE equipped_id = equipped_item_id;
+
+    -- Get the damage of the equipped item
```

Comment:
You have the right idea, but I recommend JOINing the previous SELECT and this SELECT into a single SELECT statement.